code/modules/antagonists/heretic/status_effects/void_chill.dm

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Contains the "Void Chill" status effect. Harmful debuff which freezes and slows down non-heretics
+ * Cannot affect silicons (How are you gonna freeze a robot?)
+ */
+/datum/status_effect/void_chill
+	id = "void_chill"
+	duration = 20 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/void_chill
+	status_type = STATUS_EFFECT_REFRESH //Custom code
+	on_remove_on_mob_delete = TRUE
+	remove_on_fullheal = TRUE
+	///Current amount of stacks we have
+	var/stacks
+	///Maximum of stacks that we could possibly get
+	var/stack_limit = 5
+	///icon for the overlay
+	var/mutable_appearance/stacks_overlay
+
+/datum/status_effect/void_chill/on_creation(mob/living/new_owner, new_stacks, ...)
+	. = ..()
+	RegisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(update_stacks_overlay))
+	set_stacks(new_stacks)
+	stacks_overlay = mutable_appearance('icons/effects/effects.dmi', "void_chill_oh_fuck", ABOVE_MOB_LAYER)
+	owner.update_icon(UPDATE_OVERLAYS)
+
+/datum/status_effect/void_chill/Destroy()
+	QDEL_NULL(stacks_overlay)
+	return ..()
+
+/datum/status_effect/void_chill/on_apply()
+	if(issilicon(owner))
+		return FALSE
+	RegisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(update_stacks_overlay))
+	owner.update_icon(UPDATE_OVERLAYS)
+	return TRUE
+
+/datum/status_effect/void_chill/on_remove()
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/void_chill)
+	REMOVE_TRAIT(owner, TRAIT_HYPOTHERMIC, REF(src))
+	UnregisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS)
+	owner.update_icon(UPDATE_OVERLAYS)
+
+/datum/status_effect/void_chill/tick(seconds_per_ticks)
+	owner.adjust_bodytemperature(-12 * stacks * seconds_per_ticks)
```

Comment:
```suggestion
	owner.adjust_bodytemperature(-12 KELVIN * stacks * seconds_per_ticks)
```
(`KELVIN` define doesn't actually do anything, it's just to make code easier to understand at a glance)